include/nbl/video/IGPUCommandPool.h

```diff
@@ -46,18 +46,19 @@ class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
         // static void* operator new(std::size_t size, std::align_val_t al) = delete;
         static void* operator new[](std::size_t size, std::align_val_t al) = delete;
 
+        static void operator delete[](void* ptr) = delete;
+        static void operator delete[](void* ptr, std::align_val_t al) = delete;
+        static void operator delete[](void* ptr, std::size_t sz) = delete;
+        static void operator delete[](void* ptr, std::size_t sz, std::align_val_t al) = delete;
+
         inline uint32_t getSize() const { return m_size; }
 
     protected:
+        
         static void operator delete  (void* ptr) { assert(false); }
-        static void operator delete[](void* ptr) { assert(false); }
         static void operator delete  (void* ptr, std::align_val_t al) { assert(false); }
-        static void operator delete[](void* ptr, std::align_val_t al) { assert(false); }
         static void operator delete  (void* ptr, std::size_t sz) { assert(false); }
-        static void operator delete[](void* ptr, std::size_t sz) { assert(false); }
         static void operator delete  (void* ptr, std::size_t sz, std::align_val_t al) { assert(false); }
```

Comment:
do you need all 4 ?

Reply:
look at the matching `new` that have been commented out for placement

Reply:
https://stackoverflow.com/questions/12320135/virtual-destructor-is-it-required-in-base-class-if-base-class-allocated-memory

this is what I found on stack overflow.

Reply:
> look at the matching `new` that have been commented out for placement

looks like It I tried 1 after the other they fail to compile.